contracts/governance/StkTruToken.sol

```diff
@@ -401,15 +397,46 @@ contract StkTruToken is VoteToken, StkClaimableContract, ReentrancyGuard {
      * @param token Token to claim rewards for
      */
     function _claim(IERC20 token) internal {
-        farmRewards[token].totalClaimedRewards = farmRewards[token].totalClaimedRewards.add(
-            farmRewards[token].claimableReward[msg.sender]
-        );
-        uint256 rewardToClaim = farmRewards[token].claimableReward[msg.sender];
-        farmRewards[token].claimableReward[msg.sender] = 0;
+        uint256 rewardToClaim = _claimWithoutTransfer(token);
         require(token.transfer(msg.sender, rewardToClaim));
         emit Claim(msg.sender, token, rewardToClaim);
     }
 
+    /**
+     * @dev Internal claim function that returns the transfer value
+     * Claim rewards for a specific ERC20 token to return in a uint256
+     * @param token Token to claim rewards for
+     */
+    function _claimWithoutTransfer(IERC20 token) internal returns (uint256) {
+        uint256 rewardToClaim = farmRewards[token].claimableReward[msg.sender];
+        farmRewards[token].totalClaimedRewards = farmRewards[token].totalClaimedRewards.add(rewardToClaim);
+        farmRewards[token].claimableReward[msg.sender] = 0;
+        return rewardToClaim;
+    }
+
+    /**
+     * @dev Internal stake of TRU for stkTRU from a uint256
+     * Caller is responsible for ensuring amount is transferred from a valid source
+     * @param amount Amount of TRU to stake for stkTRU
+     */
+    function _stakeWithoutTransfer(uint256 amount) internal {
+        require(amount > 0, "StkTruToken: Cannot stake 0");
+
+        if (cooldowns[msg.sender] != 0 && cooldowns[msg.sender].add(cooldownTime).add(unstakePeriodDuration) > block.timestamp) {
+            cooldowns[msg.sender] = block.timestamp;
+
+            emit Cooldown(msg.sender, block.timestamp.add(cooldownTime));
+        }
+
+        if (delegates[msg.sender] == address(0)) {
+            delegates[msg.sender] = msg.sender;
+        }
+
+        uint256 amountToMint = stakeSupply == 0 ? amount : amount.mul(totalSupply).div(stakeSupply);
+        _mint(msg.sender, amountToMint);
+        stakeSupply = stakeSupply.add(amount);
```

Comment:
it might be better to emit the Stake event here and remove it from the public functions. Same with claimWithoutTransfer. Then you know the event will always be emitted.